### ISSUE TITLE
msvc compilation

### ISSUE THREAD

**Issue by Berserker66** [viblo/pymunk#129]
So, mingwpy is dead: https://groups.google.com/forum/#!topic/mingwpy/1k_BLFPLmBI

On the plus side, the newest Chipmunk supports MSVC compilation on windows now (Microsoft visual c)

In my own test I got it to build - with some warnings:
```
1>------ Rebuild All started: Project: chipmunk, Configuration: Debug DLL x64 ------
1>cpSweep1D.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpatialIndex.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceStep.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspacestep.c(251): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceQuery.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceHash.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceDebug.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceComponent.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspacecomponent.c(65): warning C4090: 'function': different 'const' qualifiers
1>cpSpace.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspace.c(410): warning C4307: '*': integral constant overflow
1>cpSlideJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSimpleMotor.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpShape.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpRotaryLimitJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpRobust.c
1>cpRatchetJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPolyShape.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPivotJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPinJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpHashSet.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpGrooveJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpGearJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>Generating Code...
1>Compiling...
1>cpDampedSpring.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpDampedRotarySpring.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpConstraint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpCollision.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpBody.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpBBTree.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpArray.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpArbiter.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>chipmunk.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\chipmunk.c(93): warning C4028: formal parameter 2 different from declaration
1>Generating Code...
1>   Creating library x64\Debug DLL\chipmunk.lib and object x64\Debug DLL\chipmunk.exp
1>chipmunk.vcxproj -> E:\Chipmunk2D-master\msvc\vc14\chipmunk\x64\Debug DLL\chipmunk.dll
1>Done building project "chipmunk.vcxproj".
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========

```
-but I also used Visual Studio 2017, which chipmunk doesn't support yet.

And it passed a decent amount, but not all pymunk tests. The tests it failed are of the type:
KeyError: "variable '_cpBBIntersects' not found in library 'E:\\pymunk-multithread\\pymunk\\chipmunk.dll': error 0x7f"

Seems to affect polyline and marchhard/soft and a few others, but this might just be due to pymunk not being updated to handle the newest version of chipmunk.

**Comment by viblo** [comment]
Sad to see mingwpy dead. 

Previously when I looked into VS the problem was lacking support in Chipmunk. I dont think the authors test or develop on windows, so it was always waiting for community patches to make it work properly. 

Even if Im a windows person myself its quite convenient to use the same easy-to-install compiler on all the platforms :)

**Comment by viblo** [comment]
For your particular problem I think its because you are missing to define `CHIPMUNK_FFI`. Without it it doesnt include chipmunk_ffi.h in the compilation.

**Comment by Berserker66** [comment]
I was mostly looking into it, because mingwpy installation on python 3.6 and 3.7 failed. I then stumbled upon them abandoning it.

**Comment by Berserker66** [comment]
I think I managed to define the FFI. But the ffi portion may be somewhat broken (at least on the 2017 edition, I don't have the officially supported version installed)

```
1>------ Build started: Project: chipmunk, Configuration: Debug x64 ------
1>cpSweep1D.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpatialIndex.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceStep.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspacestep.c(251): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceQuery.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceHash.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceDebug.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSpaceComponent.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspacecomponent.c(65): warning C4090: 'function': different 'const' qualifiers
1>cpSpace.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\cpspace.c(410): warning C4307: '*': integral constant overflow
1>cpSlideJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpSimpleMotor.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpShape.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpRotaryLimitJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpRobust.c
1>cpRatchetJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPolyShape.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPivotJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpPinJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpHashSet.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpGrooveJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpGearJoint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>Generating Code...
1>Compiling...
1>cpDampedSpring.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpDampedRotarySpring.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpConstraint.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpCollision.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpBody.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpBBTree.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpArray.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>cpArbiter.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>chipmunk.c
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_private.h(304): warning C4090: 'function': different 'const' qualifiers
1>e:\chipmunk2d-master\src\chipmunk.c(93): warning C4028: formal parameter 2 different from declaration
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(42): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(42): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpFloat,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(43): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(43): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(44): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(44): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(45): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(45): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(46): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(46): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(47): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(47): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(48): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(48): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(49): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(49): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(50): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(50): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(51): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(51): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(52): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(52): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(53): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(53): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(54): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(54): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(55): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(55): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(56): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(56): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(57): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(57): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(58): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(58): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(59): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(59): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpVect,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(60): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(60): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(61): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(61): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpVect,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(62): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(62): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(cpVect,cpVect,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(63): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(63): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(64): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(64): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(const cpVect,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(65): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(65): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(const cpVect,const cpVect,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(67): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(67): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(68): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(68): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(69): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(69): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(70): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(70): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat,cpFloat,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(71): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(71): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat,cpFloat,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(72): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(72): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpFloat,cpFloat,cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(74): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(74): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBB (__cdecl *)(const cpFloat,const cpFloat,const cpFloat,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(75): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(75): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBB (__cdecl *)(const cpVect,const cpFloat,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(76): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(76): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBB (__cdecl *)(const cpVect,const cpFloat)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(77): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(77): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(const cpBB,const cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(78): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(78): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(const cpBB,const cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(79): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(79): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(const cpBB,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(80): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(80): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBB (__cdecl *)(const cpBB,const cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(81): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(81): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBB (__cdecl *)(const cpBB,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(82): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(82): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(83): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(83): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(84): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(84): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpBB,cpBB)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(85): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(85): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpFloat (__cdecl *)(cpBB,cpVect,cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(86): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(86): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(cpBB,cpVect,cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(87): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(87): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpVect (__cdecl *)(const cpBB,const cpVect)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(89): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(89): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(90): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(90): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'int (__cdecl *)(cpSpatialIndex *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(91): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(91): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,cpSpatialIndexIteratorFunc,void *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(92): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(92): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'cpBool (__cdecl *)(cpSpatialIndex *,void *,cpHashValue)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(93): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(93): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,void *,cpHashValue)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(94): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(94): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,void *,cpHashValue)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(95): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(95): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(96): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(96): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,void *,cpHashValue)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(97): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(97): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,void *,cpVect,cpVect,cpFloat,cpSpatialIndexSegmentQueryFunc,void *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(98): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(98): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,void *,cpBB,cpSpatialIndexQueryFunc,void *)'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(99): error C2143: syntax error: missing '{' before '*'
1>e:\chipmunk2d-master\include\chipmunk\chipmunk_ffi.h(99): warning C4047: 'initializing': 'int *' differs in levels of indirection from 'void (__cdecl *)(cpSpatialIndex *,cpSpatialIndexQueryFunc,void *)'
1>Generating Code...
1>Done building project "chipmunk.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Looks fixable, at least.

**Comment by Berserker66** [comment]
I did get it fixed, and ran a succesfull compilation that also passed the pymunk tests.
The setup module is a bit of a mess though, so I'll clean that up before submitting a pull request.

**Comment by viblo** [comment]
Another option would be to use the m2w64-toolchain mentioned in the post. Maybe it will just work directly.

**Comment by Berserker66** [comment]
I haven't checked that one out. But yeah it might be an option.

I just grabbed the visual studio project from the chipmunk source, fixed the errors, used your tips and got a working version that now also works with pymunk's setup.py and consequently passes the tests. So, if that is good enough I can pack that in a pull request.

**Comment by viblo** [comment]
I made a quick test of m2w64, and it worked fine as far as I tested (py2.7, 64bits). From previous experience I know that at minimum it must be tested on the combinations of py2/3 and 32/64bit to know for sure, but looks promising.

For VS support Im still a bit hesitant because of the additional troubles to get the compiler and the lack of support from Chipmunk (even if it works fine now). However, you are very welcome to make a PR, then at least I (and other interested persons) can look at it. (but no guarantee it will be merged)
